WIP: Add draft of experimental docs section

rachelandrew opened the pull request:
Fixes #1956 

An initial draft to demonstrate this. I've also pulled in existing content about origin trials which I think should live here rather than buried in the blog.

Ultimately this should have sections for origin trials, developer trials, features in Canary, meaning that blog posts can contain announcements and docs can be documentation and kept up to date as the trial progresses. I'm seeing these as pretty much like an MDN API Overview page, as ultimately we would move this to MDN once the feature hits stable. 

To do:

- [x] Collect launch set of docs for ongoing trials from the blog.
- [x] Add sections to main page for different trials.
- [x] Create icon for experimental docs.
- [x] Link from main docs page.

tomayac commented on site/en/docs/handbook/web-platform/index.md:
```diff
@@ -0,0 +1,72 @@
+
+Intro paragraph explaining briefly what this API is for.
+
+## Current status
+
+Add information about status, and how developers can try out the API. 
+
+## Interfaces
+
+Describe each interface that is part of the API 
+using a description list format.
+
+`interfaceName`
+: Short description.
+
+## Use cases 
+
+Change the heading above to make sense for your content, 
+however here you can add a short explanation, 
+and the core use cases for the API.
+
+## Examples
```
```suggestion
## Use cases 

Change the heading above to make sense for your content, 
however here you can add a short explanation, 
and the core use cases for the API.

## Current status

Add information about status, and how developers can try out the API. 

## Interfaces

Describe each interface that is part of the API 
using a description list format.

`interfaceName`
: Short description.

## Examples
```

tomayac replied:
Should we move **Use Cases** right after the **Outline**, so developers can early on decide if this is something they need?

rachelandrew replied:
I'd like to have the current status above it, because it's not helpful for developers to find something they need then realise they can't use it. So maybe reorder to status, then use cases, then interfaces?

tomayac replied:
Likewise sounds good to me. Deleting the suggestion then.

tomayac commented on site/en/docs/web-platform/index.md:
```diff
@@ -0,0 +1,7 @@
+description: 'Documentation for Web Platform APIs.'
+---
+
+This section contains documentation for Web Platform APIs. Some are currently in an experimental status in Chrome. For example, origin and developer trials, or features only found in Canary. Others have shipped in stable Chrome, however are unimplemented or currently under development in other engines.
```
```suggestion
This section contains documentation for Web Platform APIs. Some are currently in an experimental status in Chromium. For example, origin and developer trials, or features only found in Canary. Others have shipped in stable Chromium, however are unimplemented or currently under development in other engines.
```

tomayac replied:
Not sure about this one. We might suggest that not all Chromium-derivates enable an API if it's supported in the engine. Example: Brave chose to disable the File System Access API despite the engine supporting it.

rachelandrew replied:
Yeah, I was thinking that this is really Chrome, or at least we can be sure they are available in Chrome.

With shipped things we should be clear about status, especially if we know that some Chromium-based browsers haven't shipped. So I think I'll leave this as Chrome based on that?

tomayac replied:
Let's leave it at Chrome. That's one advantage of writing on DCC :-)

rachelandrew commented:
> Should we move Use Cases right after the Outline, so developers can early on decide if this is something they need?

I thought I'd replied to this but apparently not! I want to make sure the status is prioritized. As one of the things that annoys people is when they think something will solve their problem then realize it's only a trial. I've moved Use Cases up under Status though.

> Not sure about this one. We might suggest that not all Chromium-derivates enable an API if it's supported in the engine. Example: Brave chose to disable the File System Access API despite the engine supporting it.

Thinking about it I'm not sure we need say anything. So I've removed that.
